### PR TITLE
feat: add Franca examples and docs

### PR DESCRIPTION
**Pull request by lorinczszabolcs**
Adds the Franca examples (PyTorch, Lightning, Lightning distributed), the `franca.rst` docs page in the models toctree, and the regenerated notebooks for #2010. Updates the README methods table with a Franca entry.

The example wires up the full method: an EMA teacher, the matryoshka projection head for both the DINO image-level and iBOT patch-level branches, cyclic block masking for the iBOT branch, and the two matryoshka losses summed per nesting level plus the KoLeo regularizer. Scaled down to run on a single GPU (fewer prototypes, a small backbone, few epochs), like the other examples.

Follow up of #2064 

## Summary

- Added Franca examples for PyTorch, Lightning, distributed Lightning, and regenerated notebooks.
- Added `FrancaDINOLoss`, `FrancaIBOTPatchLoss`, `FrancaProjectionHead`, and `random_cyclic_block_mask`.
- Added tests for the new losses, projection head, and masking utility.
- Added Franca documentation, README links, and the models toctree entry.

Test execution results were not provided.

### CODE REVIEW

**Comment by lorinczszabolcs** [comment]
[franca.pdf](https://github.com/user-attachments/files/32233123/franca.pdf)

File: docs/source/examples/franca.rst [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+.. tabs::
+    .. tab:: PyTorch
+
+        .. image:: /_static/colab-badge.svg
+            :target: https://colab.research.google.com/github/lightly-ai/lightly/blob/master/examples/notebooks/pytorch/franca.ipynb
+
+        This example can be run from the command line with::
+
+            python lightly/examples/pytorch/franca.py
```
[the remark this answers is not in the document]
Comment (lorinczszabolcs) [reply docs/source/examples/franca.rst:40]: I kept the `lightly/` prefix to match every other method page (for example `dinov2.rst` documents `python lightly/examples/pytorch/dinov2.py`). Changing only this page would make it inconsistent with the rest of the docs. If the prefix should change, that is a repo-wide docs update rather than a Franca-specific one.

File: examples/notebooks/pytorch_lightning_distributed/franca.ipynb [hunk 2]
```diff
@@ -0,0 +1,450 @@
...
+    "    def configure_optimizers(self):\n",
+    "        optim = AdamW(self.parameters(), lr=0.001)\n",
+    "        return optim\n",
+    "\n",
+    "    def on_before_optimizer_step(self, optimizer: AdamW, *args) -> None:\n",
+    "        # Optionally zero out the learning rate of the last layer.\n",
+    "        if self.current_epoch < 1:\n",
+    "            for param_group in optimizer.param_groups:\n",
+    "                if \"last_layer\" in param_group:\n",
+    "                    param_group[\"lr\"] = 0.0\n",
```
[the remark this answers is not in the document]
Comment (lorinczszabolcs) [reply examples/notebooks/pytorch_lightning_distributed/franca.ipynb:324]: This mirrors the existing `dinov2.py` Lightning examples, where the epoch-0 last-layer freeze is also inert because the optimizer uses a single default parameter group. The benchmark implements the freeze properly with named parameter groups. I kept the example consistent with the other methods rather than diverging here.

File: examples/notebooks/pytorch_lightning_distributed/franca.ipynb [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+   "source": [
+    "# Train with DDP and use Synchronized Batch Norm for a more accurate batch norm\n",
+    "# calculation. Distributed sampling is also enabled with replace_sampler_ddp=True.\n",
+    "trainer = pl.Trainer(\n",
+    "    max_epochs=50,\n",
+    "    devices=\"auto\",\n",
+    "    accelerator=\"gpu\",\n",
+    "    strategy=\"ddp_find_unused_parameters_true\",\n",
+    "    sync_batchnorm=True,\n",
+    "    use_distributed_sampler=True,  # or replace_sampler_ddp=True for PyTorch Lightning <2.0\n",
```
[the remark this answers is not in the document]
Comment (lorinczszabolcs) [reply examples/notebooks/pytorch_lightning_distributed/franca.ipynb:435]: This follows the existing `dinov2.py` distributed example, which also uses `use_distributed_sampler=True` and documents the 1.x alternative in the inline comment. Dropping 1.x support in the examples would be a repo-wide change rather than a Franca-specific one.
